File: CSharpCodeBase/components/thing/thingdamagereceiver.cs
using System;
 
 using System.Collections.Generic;
 
 using System.Linq;
 
 using System.Text;
 
 using UnityEngine;
 
 using System.Collections;
 
 using MainGame.core;
 
 using UnityEngine.EventSystems;
 
 namespace MainGame{
 public class thingdamagereceiver :component{
 var ThingDamageReceiverComponent = Class({__includes=Component, init=function(self, entity)
       Component.init(self, entity);
 }})
 public void ThingDamageReceiverComponent:OnApplyDamage(damageData){
   var isDeath = this.entity.health:Decrease(damageData.summary) == 0;
   if(not isDeath  ){ 
     this.entity:Hit(damageData);
   }else{
     this.entity:DestroyThing(damageData);
   }
 }
 Component}}

File: CSharpCodeBase/components/thing/thingdamagereceiver.lua
local Class = require 'utils.hump.class'
local Component = require 'components.component'

local ThingDamageReceiverComponent = Class({__includes=Component, init=function(self, entity)
      Component.init(self, entity)
end})

function ThingDamageReceiverComponent:OnApplyDamage(damageData)
  local isDeath = self.entity.health:Decrease(damageData.summary) == 0
  if not isDeath then
    self.entity:Hit(damageData)
  else
    self.entity:DestroyThing(damageData)
  end
end

return ThingDamageReceiverComponent